google-cloud-trace/lib/google/cloud/trace/v1/credentials.rb

```diff
@@ -0,0 +1,34 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "googleauth"
+
+module Google
+  module Cloud
+    module Trace
+      module V1
+        class Credentials < Google::Auth::Credentials
+          SCOPE = [
+            "https://www.googleapis.com/auth/cloud-platform",
```

Comment:
I am not a fan of how the generator chooses scopes here. This list is all the scopes that _could_ be used, but by specifying them here they all _must_ be used. Users should either auth with the `cloud-platform` scope, or the `trace.append` scope, or the `trace.readonly` scope, depending on how they want to use the library. But now the default behavior is that they must have access to all these scopes.

This is why only `cloud-platform` appears in the previous hand-coded credentials file. That seems like a sensible default scope.

Reply:
/cc @jbolinger Is there anything we can do to make the default scope better?

Reply:
@blowmage of course - we can do whatever we'd like. I agree that all by default is probably the best strategy.

However, I asked about this quite some time ago and I believe that I got some pushback (can't recall the details). I think the argument was that the current way makes it easier to get started, but I'm not sure I buy it. I'd be more than happy to make a change but I think we'll need input from the Yoshi side. @dazuma WDYT? Do you know if this has come up in other languages?

Reply:
I asked around a bit in Yoshi. There is some disagreement (and confusion) on it, especially for cloud APIs where cloud-platform generally covers the needed endpoints. Given this, I'm hesitant to make a change from what we're currently doing in the hand-written credentials files, since we know it is currently working for users. Meaning, we should use only `cloud-platform` by default, until we have a compelling reason to change it.

Reply:
Changed synth to reduce the scopes down to `cloud-platform` only, and regenerated.